pkg/provisioner/ironic/ironic.go

```diff
@@ -974,7 +988,31 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return p.startProvisioning(ironicNode, checksum, hostConf)
 
 	case nodes.Manageable:
-		return p.startProvisioning(ironicNode, checksum, hostConf)
+		if false == p.host.Spec.CleanFlag {
```

Comment:
It looks like we're storing this new state in the input fields for the API, instead of the status fields. Why is that? Maybe because it's addressing the old comment about unconditionally entering cleaning? Do we expect the user to ever set CleanFlag themselves? If it's just a way to track state ourselves, can we do that with a new value in the state machine that we already have? Or can we look at the node state in ironic?